controller-server/src/main/java/com/yahoo/vespa/hosted/controller/LockedTenant.java

```diff
@@ -154,8 +155,11 @@ public CloudTenant get() {
 
         public Cloud withDeveloperKey(PublicKey key, Principal principal) {
             BiMap<PublicKey, Principal> keys = HashBiMap.create(developerKeys);
+            principal = new SimplePrincipal(principal.getName());
```

Comment:
Ideally we should change the type of the `BiMap` to also be `SimplePrincipal`, if someone changes deserialization code, this could stop working again?